lib/glific/groups/wa_groups.ex

```diff
@@ -39,12 +51,95 @@ defmodule Glific.Groups.WAGroups do
     end
   end
 
+  @spec get_group_details(map(), WAManagedPhone.t()) :: [map()]
   defp get_group_details(%{"data" => groups}, wa_managed_phone) when is_list(groups) do
     Enum.map(groups, fn group ->
-      %{name: group["name"], bsp_id: group["id"], wa_managed_phone_id: wa_managed_phone.id}
+      %{
+        name: group["name"],
+        bsp_id: group["id"],
+        wa_managed_phone_id: wa_managed_phone.id,
+        participants: group["participants"] || [],
+        admins: group["admins"]
+      }
     end)
   end
 
+  @doc false
+  @spec sync_wa_groups_with_contacts(list(), non_neg_integer()) :: :ok | {:error, any()}
+  def sync_wa_groups_with_contacts(group_details, org_id) do
+    Enum.each(group_details, fn group ->
+      wa_group_id = wa_group_id(group.bsp_id)
+      admin_phone_number = Enum.at(group.admins, 0) |> phone_number()
+
+      Ecto.Multi.new()
+      |> delete_existing_contacts(wa_group_id)
+      |> add_wa_contact(group, wa_group_id, admin_phone_number, org_id)
+      |> Repo.transaction()
+      |> handle_transaction_result()
```

Comment:
omit writing query and try using functions in module file and add functions there